language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/DamlDataTypeGen.scala

```diff
@@ -134,11 +135,56 @@ object DamlRecordOrVariantTypeGen {
           $valueInstanceExpr"""
     }
 
+    /**
+      *  The generated class for a DAML enum type contains:
+      *  - the definition of a "Value" trait
+      *  - the definition of a _case object_ for each constructor of the DAML enum
+      *  - A type class instance (i.e. implicit object) for serializing/deserializing
+      *    to/from the ArgumentValue type (see typed-ledger-api project)
+      */
+    def toScalaDamlEnumType(constructors: List[Ref.Name]): (Set[Tree], (Tree, Tree)) = {
+      val className = damlScalaName.name.capitalize
+
+      val klass =
+        q"""
+          sealed abstract class ${TypeName(className)}(
+            override val constructor: String,
+            override val index: Int
+          ) extends ${tq"$domainApiAlias.EnumRef"} {
+            ..$rootClassChildren
+          }"""
+
+      val (imports, companionObject) = constructors match {
```

Comment:
For whatever reason, the `::` pattern is super brittle.

```suggestion
      val (imports, companionObject) = (constructors: List[Ref.Name]) match {
```